Mathlib/FieldTheory/Galois/Basic.lean

```diff
@@ -516,18 +516,18 @@ instance (priority := 100) IsAlgClosure.isGalois (k K : Type*) [Field k] [Field
 
 end IsAlgClosure
 
-section IsQuadraticExtension
+section IsQuadraticAlgebra
 
 /--
 A quadratic separable extension is Galois.
 -/
-instance (F K : Type*) [Field F] [Field K] [IsQuadraticExtension F K] [Algebra.IsSeparable F K] :
-    IsGalois F K where
+instance (F K : Type*) [Field F] [Field K] [Algebra F K] [IsQuadraticAlgebra F K]
+    [Algebra.IsSeparable F K] : IsGalois F K where
 
 /--
 A quadratic extension has cyclic Galois group.
 -/
-instance (F K : Type*) [Field F] [Field K] [h : IsQuadraticExtension F K] :
+instance (F K : Type*) [Field F] [Field K] [Algebra F K] [h : IsQuadraticAlgebra F K] :
```

Comment:
I'm not a big fan of `[h : IsQuadraticAlgebra F K]`. I guess that the situation is this: either it shouldn't be a class, and you can call it h, or it is a class, and then there should be some lemma `finrank_eq_two` which takes F and K explicitly and finds the instance with typeclass inference.

Reply:
I added a lemma `finrank_eq_two` so that there is no need anymore to use name the hypothesis.